portality/lib/es_snapshots.py

```diff
@@ -0,0 +1,84 @@
+from datetime import datetime, timedelta
+import requests
+from portality.core import app
+
+
+class BadSnapshotMetaException(Exception):
+    pass
+
+
+class TodaySnapshotMissingException(Exception):
+    pass
+
+
+class FailedSnapshotException(Exception):
+    pass
+
+
+class SnapshotDeleteException(Exception):
+    pass
+
+
+class ESSnapshot(object):
+    def __init__(self, snapshot_json):
+        self.data = snapshot_json
+        self.name = snapshot_json['snapshot']
+        self.state = snapshot_json['state']
+        self.datetime = datetime.utcfromtimestamp(snapshot_json['start_time_in_millis'] / 1000)
```

Comment:
For all these where we read a property from the input, would it be safer to use `.get` rather than accessing the property directly.  Then its value can be defaulted, or if it is missing it won't necessarily break the class.  Or do we want it to break the class during construction?

Reply:
It intentionally fails if these aren't present, and there's no sensible default for representing a genuine snapshot record. If these are missing, something isn't right. That's caught on line 60.